docs/new/consume/stream/go.md

```diff
@@ -0,0 +1,138 @@
+
+	endpoint := os.Args[1] // 3.
+	manifestPath := os.Args[2] // 4.
+
+	outputModuleName := sink.InferOutputModuleFromPackage // 5.
+	if len(os.Args) == 4 {
+		outputModuleName = os.Args[3] // 6.
+	}
+
+	params := ([]string)(nil) // 7.
+
+    // ...code omitted...
+}
+```
+1. The output type of the Substreams module (i.e. the Protobuf output). In this example, the `db_out` emits `DatabaseChanges`. This will depend on the Substreams you are executing and the output type it emits.
+2. Verify the number of parameters passed to the Go program. In this case, the endpoint and the manifest (package) are mandatory, while the module name is not.
+3. Store the endpoint in a variable.
+4. Store the manifest in a variable.
+5. If the module name is NOT present in the parameters, this option tries to infer the module name from the manifest.
+6. If the module is present in the parameters, then you store it in a variable.
+7. Set the parameters for the Substreams module.
+These are [parameters of the Substreams](../../develop/parameterized-modules.md) itself, do not confuse them with the parameters of the Go program.
+
+Once you have all the information needed, you can create the actual Go structs from the string variables:
+
+```go
+// ...code omitted...
+
+func main() {
+    // ...code omitted...
+
+    pkg, outputModule, outputHashModule, blockRange, err := sink.ReadManifestAndModuleAndBlockRange(
+            manifestPath,
+            "mainnet", // 1.
+            params,
+            outputModuleName,
+            expectedOutputModuleType, // 2.
+            false,
+            ":", // 3.
+            zlog,
+        )
+    
+    // ...code omitted...
+}
+```
+1. Equivalent to the `network` parameter of the CLI. Specify the network to use for params and initialBlocks, overriding the `network`` field in the Substreams package.
+2. The output type emitted by the module.
+3. The range block to execute. `:` means from the start of the chain up to the latest block.
+
+The `ReadManifestAndModuleAndBlockRange` will generate several Go structs collected in their corresponding variables (`pkg`, `outputModule`...), which will be used to initialize the sink:
+
+```go
+// ...code omitted...
+
+func main() {
+    // ...code omitted...
+
+    sinker, err := sink.New(
+            sink.SubstreamsModeProduction, // 1.
+            pkg,
+            outputModule,
+            outputHashModule,
+            client.NewSubstreamsClientConfig(endpoint, os.Getenv("SUBSTREAMS_API_TOKEN"), false, false), // 2.
+            zlog,
+            tracer,
+            sink.WithBlockRange(blockRange), // 3.
+        )
+    
+    // ...code omitted...
+}
+```
+1. Run the Substreams in _production_ mode.
+2. Create a Substreams client, which includes the endpoint and the authentication token.
+This will also work if you have a system environment variable, `SUBSTREAMS_API_TOKEN`, with a valid Substreams token.
```

Comment:
Should we mention that `false, false` parameters represent `insecure, plainText` parameters